tests/functional/test_proxy.py

```diff
@@ -0,0 +1,89 @@
+import ssl
+from pathlib import Path
+from typing import Any, Dict
+
+import proxy
+import pytest
+from proxy.http.proxy import HttpProxyBasePlugin
+
+from tests.conftest import CertFactory
+from tests.lib import PipTestEnvironment, TestData
+from tests.lib.server import (
+    authorization_response,
+    make_mock_server,
+    package_page,
+    server_running,
+)
+
+
+class AccessLogPlugin(HttpProxyBasePlugin):
+    def on_access_log(self, context: Dict[str, Any]) -> None:
+        print(context)
+
+
+@pytest.mark.network
+def test_proxy_overrides_env(script: PipTestEnvironment) -> None:
+    with proxy.Proxy(
+        port=8899,
+        num_acceptors=1,
+    ), proxy.Proxy(plugins=[AccessLogPlugin], port=8888, num_acceptors=1):
+        script.environ["http_proxy"] = "127:0.0.1:8888"
+        script.environ["https_proxy"] = "127:0.0.1:8888"
```

Comment:
I am looking at the code to figure out reasons behind Windows test failures.  Shouldn't this be like `127.` instead of `127:`.  Unsure if this even can be the root cause, but worth pointing out :)